### PR TITLE
object_du OK

### PR DESCRIPTION
**Pull request by AngheloAlf**
`object_du` (Darunia) is completely decompiled.

I also decompiled a few scenes from `spot18` (Goron City) and added its references to `ovl_En_Du`. There is a problem with one scene. See this [PR](https://github.com/zeldaret/ZAPD/pull/91).

### CODE REVIEW

**Comment by fig02** [comment]
now that the accompanying actor is getting merged youll need to update the actor to use these symbols

File: assets/xml/objects/object_du.xml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        <Animation Name="gDaruniaCreditsHitBreastAnim" Offset="0x6104"/>
+        <Animation Name="gDaruniaCreditsIdleAnim" Offset="0x67CC"/>
+
+
+        <!-- Darunia's eyes textures -->
+        <!-- TODO: Rename darunia_eye_half1 and darunia_eye_half2 to something more clear -->
+        <Texture Name="gDaruniaEyeOpenTex" OutName="darunia_eye_open" Format="ci8" Width="32" Height="32" Offset="0x8080"/>
+        <Texture Name="gDaruniaEyeHalf1Tex" OutName="darunia_eye_half1" Format="ci8" Width="32" Height="32" Offset="0x8480"/>
+        <Texture Name="gDaruniaEyeShutTex" OutName="darunia_eye_shut" Format="ci8" Width="32" Height="32" Offset="0x8880"/>
+        <Texture Name="gDaruniaEyeHalf2Tex" OutName="darunia_eye_half2" Format="ci8" Width="32" Height="32" Offset="0xA540"/>
```
Comment (fig02) [inline assets/xml/objects/object_du.xml:47]: you can probably use the code to determine which one is for "opening" and "closing".
some actors use the same half eye tex for both, and some have it separated like this

File: assets/xml/objects/object_du.xml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+        <!-- Darunia's mouth textures -->
+        <Texture Name="gDaruniaMouthSeriousTex" OutName="darunia_mouth_serious" Format="ci8" Width="64" Height="32" Offset="0x8C80"/>
+        <Texture Name="gDaruniaMouthTeethsTex" OutName="darunia_mouth_teeths" Format="ci8" Width="64" Height="32" Offset="0x9D40"/>
```
Comment (fig02) [inline assets/xml/objects/object_du.xml:51]: "teeths" -> "teeth"
though id try to describe the facial expression if possible, grinning maybe?

**Comment by AngheloAlf** [comment]
Done!

File: assets/xml/objects/object_du.xml [hunk 3]
```diff
@@ -47,7 +47,7 @@
 
         <!-- Darunia's mouth textures -->
         <Texture Name="gDaruniaMouthSeriousTex" OutName="darunia_mouth_serious" Format="ci8" Width="64" Height="32" Offset="0x8C80"/>
-        <Texture Name="gDaruniaMouthTeethsTex" OutName="darunia_mouth_teeths" Format="ci8" Width="64" Height="32" Offset="0x9D40"/>
+        <Texture Name="gDaruniaMouthGriningTex" OutName="darunia_mouth_grinning" Format="ci8" Width="64" Height="32" Offset="0x9D40"/>
```
Comment (fig02) [inline assets/xml/objects/object_du.xml]: missed an `n` in the variable name, otherwise good to go

**Review by fig02 — APPROVED** [review]

File: assets/xml/scenes/overworld/spot18.xml [hunk 4]
```diff
@@ -1,6 +1,12 @@
 <Root>
     <File Name="spot18_scene">
         <Scene Name="spot18_scene">
+            <CutsceneHint Name="gGoronCityDaruniaCorrectCs" Offset="0x59E0"/>
+
+            <!-- For some reason, this name (`gGoronCityDarunia01Cs`) is ignored, and `spot18_sceneCutsceneData0x006930` is used instead. -->
+            <CutsceneHint Name="gGoronCityDarunia01Cs" Offset="0x6930"/>
```
Comment (fig02) [inline assets/xml/scenes/overworld/spot18.xml:7]: we could try to take a look at zap and see if this is an easy fix. would be good to sort this out before merging this one specifically

Comment (AngheloAlf) [reply assets/xml/scenes/overworld/spot18.xml:7]: Good idea. I'll take a look in ZAPD

Comment (AngheloAlf) [reply assets/xml/scenes/overworld/spot18.xml:7]: I think I found a fix. I'll do a PR in ZAPD.

**Comment by Roman971** [comment]
I could be wrong but I think this needs to be updated to use the proper cs symbol now, since the ZAPD fix was merged

**Comment by AngheloAlf** [comment]
I actually had forgotten about this problem :sweat_smile: . It should be fixed now.

**Review by fig02 — APPROVED** [review]

File: assets/xml/objects/object_du.xml [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+        <!-- Darunia's nose textures -->
+        <Texture Name="gDaruniaNoseSeriousTex" OutName="darunia_nose_serious" Format="ci8" Width="8" Height="8" Offset="0x7FC0"/>
+        <Texture Name="gDaruniaNoseHappyTex" OutName="darunia_nose_happy" Format="ci8" Width="8" Height="8" Offset="0xB140"/>
+
+
+        <!-- Palettes -->
+        <Texture Name="gDaruniaPalette_6EC0" OutName="palette_6EC0" Format="rgb5a1" Width="16" Height="16" Offset="0x6EC0"/>
+        <Texture Name="gDaruniaPalette_7AF8" OutName="palette_7AF8" Format="rgb5a1" Width="29" Height="8" Offset="0x7AF8"/>
+        <Texture Name="gDaruniaPalette_7CC8" OutName="palette_7CC8" Format="rgb5a1" Width="21" Height="12" Offset="0x7CC8"/>
+        <Texture Name="gDaruniaPalette_7980" OutName="palette_7980" Format="rgb5a1" Width="47" Height="4" Offset="0x7980"/>
```
Comment (Roman971) [inline assets/xml/objects/object_du.xml:63]:
```suggestion
        <Texture Name="gDaruniaTLUT_6EC0" OutName="tlut_6EC0" Format="rgb5a1" Width="16" Height="16" Offset="0x6EC0"/>
        <Texture Name="gDaruniaTLUT_7AF8" OutName="tlut_7AF8" Format="rgb5a1" Width="29" Height="8" Offset="0x7AF8"/>
        <Texture Name="gDaruniaTLUT_7CC8" OutName="tlut_7CC8" Format="rgb5a1" Width="21" Height="12" Offset="0x7CC8"/>
        <Texture Name="gDaruniaTLUT_7980" OutName="tlut_7980" Format="rgb5a1" Width="47" Height="4" Offset="0x7980"/>
```

**Review by Roman971 — APPROVED** [review]

**Pull request merged**